src/baseline.rs

```diff
@@ -7,22 +7,232 @@ use crate::dump::RustDocCommand;
+}
+
+impl<'a> CrateSource<'a> {
+    fn get_name(&self) -> String {
+        match self {
+            Self::Registry { crate_ } => crate_.name().to_string(),
+            Self::ManifestPath { .. } => unimplemented!(),
+        }
+    }
+
+    fn get_version(&self) -> String {
+        match self {
+            Self::Registry { crate_ } => crate_.version().to_string(),
+            Self::ManifestPath { .. } => unimplemented!(),
+        }
+    }
+}
+
+/// To get the rustdoc of the baseline, we first create a placeholder project somewhere
+/// with the baseline as a dependency, and run `cargo rustdoc` on it.
+fn create_placeholder_rustdoc_manifest(
+    crate_source: &CrateSource,
+) -> anyhow::Result<cargo_toml::Manifest<()>> {
+    use cargo_toml::*;
+
+    Ok(Manifest::<()> {
+        package: {
+            let mut package = Package::new("rustdoc", "0.0.0");
+            package.publish = Inheritable::Set(Publish::Flag(false));
+            Some(package)
+        },
+        workspace: Some(Workspace::<()>::default()),
+        lib: {
+            let product = Product {
+                path: Some("lib.rs".to_string()),
+                ..Product::default()
+            };
+            Some(product)
+        },
+        dependencies: {
+            let project_with_features: DependencyDetail = match crate_source {
+                CrateSource::Registry { crate_ } => DependencyDetail {
+                    // We need the *exact* version as a dependency, or else cargo will
+                    // give us the latest semver-compatible version which is not we want.
+                    // Fixes: https://github.com/obi1kenobi/cargo-semver-checks/issues/261
+                    version: Some(format!("={}", crate_.version())),
+                    features: crate_features::get_all_crate_features_from_registry(crate_),
+                    ..DependencyDetail::default()
+                },
+                CrateSource::ManifestPath { path, .. } => DependencyDetail {
+                    path: Some(
+                        // The manifest will be saved in some other directory,
+                        // so for convenience, we're using absolute paths.
+                        path.canonicalize()
+                            .context("failed to canonicalize manifest path")?
+                            .to_str()
+                            .context("manifest path is not valid UTF-8")?
+                            .to_string(),
+                    ),
+                    features: crate_features::get_all_crate_features_from_manifest(path),
+                    ..DependencyDetail::default()
+                },
+            };
+            let mut deps = DepsSet::new();
+            deps.insert(
+                match crate_source {
+                    CrateSource::Registry { crate_ } => crate_.name().to_string(),
+                    CrateSource::ManifestPath { name, path: _ } => name.to_string(),
+                },
```

Comment:
This is the same function as defined above. Assuming you accept my suggestion there, this could be just:
```suggestion
                crate_source.name().to_string(),
```

Reply:
I just did it a few seconds before you reviewed it :)